app/controllers/v3/application_controller.rb

```diff
@@ -16,6 +16,14 @@ def unprocessable!(message)
   def unauthorized!
     raise CloudController::Errors::ApiError.new_from_details('NotAuthorized')
   end
+
+  def method_not_allowed!(method, obj)
```

Comment:
lIRC we had at least one legit case for a 405 - but I'll look at each of the cases you pointed out and be sure we're staying on the reservation.